server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -288,10 +287,15 @@ protected void announceExistence(HostAndPort clientAddress)
     LockWatcher lw = new LockWatcher() {
       @Override
       public void lostLock(final LockLossReason reason) {
-        Halt.halt(1, () -> {
-          LOG.error("Compactor lost lock (reason = {}), exiting.", reason);
-          gcLogger.logGCInfo(getConfiguration());
-        });
+        if (isShutdownRequested()) {
```

Comment:
If a compactor process looses it lock anything it was working on will may be cleaned up in the metadata table and after that cleanup any work the compactor completes completes would be discarded.  So  may want to always halt when the lock is lost.

Reply:
Could try to detemine if this happening in the correct time window like the tserver.  However its not as important for the compactor as the tserver.  If its still compacting, did not delete its own lock, and its lock is gone then there is a chance that further work done is pointless but its not harmful.

Reply:
Addressed in 0d5f014